core/sql/optimizer/EncodedKeyValue.cpp

```diff
@@ -66,15 +66,22 @@ NAString * getMinMaxValue(desc_struct * column,
   if (NAColumn::createNAType(&column->body.columns_desc, NULL, type, NULL))
     return NULL;
   
-  Lng32 buflen = type->getTotalSize();
+  Lng32 buflen = type->getTotalSize(); 
   char * buf = new char[buflen]; // deleted at the end of this method
 
+  if (type->supportsSQLnullPhysical())
+  {
+      Lng32 nullHdrSize = type->getSQLnullHdrSize();
+      buf[0] = buf[1] = '\0';
```

Comment:
Seems like the code should zero out nullHdrSize bytes (for example, one can use memset, or alternatively a loop).